compiler/noirc_printable_type/src/lib.rs

```diff
@@ -266,14 +277,13 @@ impl std::fmt::Display for PrintableValueDisplay {
 /// A hex string's length needs to be even to decode into bytes, as two digits correspond to
 /// one byte.
 fn format_field_string(field: FieldElement) -> String {
-    if field.is_zero() {
-        return "0x00".to_owned();
-    }
-    let mut trimmed_field = field.to_hex().trim_start_matches('0').to_owned();
-    if trimmed_field.len() % 2 != 0 {
-        trimmed_field = "0".to_owned() + &trimmed_field;
-    }
-    "0x".to_owned() + &trimmed_field
+    let trimmed = field.to_hex().trim_start_matches('0').to_owned(); // empty when field is 0
+
+    // select "00" when field is 0, "" when trimmed field length is even, and "0" when the length is odd
+    let chars = ["00", "", "0"];
+    let index = (!trimmed.is_empty() as usize) + (trimmed.len() & 1);
+
+    format!("0x{}{}", chars[index], trimmed)
```

Comment:
Same here. Also, can we move this function to somewhere where it can be shared across both crates?

Reply:
Yes, sure, but where? one is in `compiler`, the other in `tooling`